### PR TITLE
Node Interface

### PR DESCRIPTION
**Pull request by xuorig**
cc @cjoudrey

### CODE REVIEW

File: app/models/graph/schema.rb [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
       Graph::Schema.types.values.find { |type| type.name == obj.class.name }
     end
+
+    id_from_object ->(object, type_definition, query_ctx) {
+      URI::GID.build(app: GlobalID.app, model_name: type_definition.name, model_id: object.id)
+    }
+
+    object_from_id ->(id, query_ctx) {
+      gid = GlobalID.parse(id)
+      Object.const_get(gid.model_name).find(gid.model_id)
```
Comment (xuorig) [inline app/models/graph/schema.rb]: Can also be better if we have model on metadata, think it would be cool

File: app/models/graph/schema.rb [hunk 2]
```diff
@@ -1,8 +1,18 @@
...
       Graph::Schema.types.values.find { |type| type.name == obj.class.name }
     end
+
+    id_from_object ->(object, type_definition, query_ctx) do
+      URI::GID.build(app: GlobalID.app, model_name: type_definition.name, model_id: object.id)
+    end
+
+    object_from_id ->(id, query_ctx) do
+      gid = GlobalID.parse(id)
+      Object.const_get(gid.model_name).find(gid.model_id)
```
Comment (cjoudrey) [inline app/models/graph/schema.rb]: Maybe I'm a bit paranoid, but think we should do this?

```ruby
type_name = uri.model_name.to_sym

if uri.app != GlobalID.app || !Graph::Types.constants.include?(type_name)
  raise GraphQL::ExecutionError, "Invalid gid: #{gid}"
end

Object.const_get(gid.model_name).find(gid.model_id)
```

File: app/models/graph/schema.rb [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
       Graph::Schema.types.values.find { |type| type.name == obj.class.name }
     end
+
+    id_from_object ->(object, type_definition, query_ctx) do
+      URI::GID.build(app: GlobalID.app, model_name: type_definition.name, model_id: object.id)
+    end
+
+    object_from_id ->(id, query_ctx) do
+      gid = GlobalID.parse(id)
+      possible_types = query_ctx.warden.possible_types(GraphQL::Relay::Node.interface)
```
Comment (xuorig) [inline app/models/graph/schema.rb:15]: cc @cjoudrey good point,  I think this is more even more accurate ^

**Comment by xuorig** [comment]
merging to keep going forward, let me know if you find anything else, will fix

File: app/models/graph/schema.rb (hunk 3 above)
Comment (cjoudrey) [reply app/models/graph/schema.rb:15]: Nice! Forgot about this.

**Pull request merged**